lib/rules/sort-alternatives.ts

```diff
@@ -156,29 +163,72 @@ function approximateLexicographicallySmallest(
     return getLexicographicallySmallestFromCharSets(prefix)
 }
 
+function getLexicographicallySmallestFromAlternative(
+    alternative: Alternative,
+    parser: JS.Parser,
+    flags: ReadonlyFlags,
+): Word | undefined
+function getLexicographicallySmallestFromAlternative(
+    alternative: StringAlternative,
+    parser: JS.Parser,
+    flags: ReadonlyFlags,
+): Word
 /**
  * If defined, this will return the lexicographically smallest string accepted
  * by the given alternative (ignoring assertions).
  */
 function getLexicographicallySmallestFromAlternative(
-    alternative: Alternative,
+    alternative: Alternative | StringAlternative,
     parser: JS.Parser,
     flags: ReadonlyFlags,
 ): Word | undefined {
-    const { elements } = alternative
-    if (isOnlyCharacters(elements)) {
+    if (
+        alternative.type === "StringAlternative" ||
+        hasOnlyCharacters(alternative, flags)
+    ) {
         // fast path to avoid converting simple alternatives into NFAs
         const smallest: Word = []
-        for (const e of elements) {
-            // FIXME: TS Error
-            // @ts-expect-error -- FIXME
-            const cs = toCharSet(e, flags)
+        for (const e of alternative.elements) {
+            const cs = toUnicodeSet(e, flags).chars
             if (cs.isEmpty) return undefined
             smallest.push(cs.ranges[0].min)
         }
         return smallest
     }
 
+    if (isOnlyCharacterElements(alternative.elements)) {
+        const reversedElements = [...alternative.elements].reverse()
+        const smallest: Word = []
+        for (const e of reversedElements) {
+            const us = toUnicodeSet(e, flags)
+            if (us.isEmpty) return undefined
+            if (us.accept.isEmpty) {
+                smallest.unshift(us.chars.ranges[0].min)
+            } else {
+                const words: ReadonlyWord[] = [
+                    ...(us.chars.isEmpty ? [] : [[us.chars.ranges[0].min]]),
+                    ...us.accept.words,
```

Comment:
Same issue as in #588. Fixed with #604.

Reply:
I've added the same fix to this PR. (But there may be conflicts)